infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/actions/RestoreRedundancyMergeSegments.java

```diff
@@ -195,6 +206,28 @@ Layout restoreWithBackOff(CorfuRuntime runtime, StateTransferManager transferMan
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Send the last transferred address as the committed tail to the target log unit.
+     * This is required to prevent loss of committed tail after state transfer finishes
+     * and then all other log units failed and auto commit service is paused.
+     */
+    private void transferCommittedTail(CorfuRuntime runtime, Layout layout, long committedTail) {
+        final int maxRetry = 3;
+        LogUnitClient logUnitClient = runtime.getLayoutView()
+                .getRuntimeLayout(layout)
+                .getLogUnitClient(currentNode);
+
+        for (int i = 0; i < maxRetry; i++) {
+            try {
+                CFUtils.getUninterruptibly(logUnitClient.updateCommittedTail(committedTail),
+                        TimeoutException.class, NetworkException.class);
```

Comment:
Does it only update current node's committedTail? Should it also update source node's ct?

Reply:
Should we add a log info/debug here to indicate transfer CT succeed? 
updateCommittedTail in LogUnitServer's log is in trace level, and it would be better for us to know retry exhausts or not

Reply:
We only need to update current node's committed tail to prevent an edge case that losing all committed tails. We don't update the source because we don't want the state transfer to have too much dependency on the committed tail, if the source is not quite responsive, the state transfer on current node can not be finished.